cpp/src/decisiontree/decisiontree_impl.cuh

```diff
@@ -174,28 +175,54 @@ void build_treelite_tree(TreeBuilderHandle tree_builder,
           TreeliteTreeBuilderCreateNode(tree_builder, node_id + 2));
 
         // Set node from current level as numerical node. Children IDs known.
+        ValueHandle threshold;
+        TREELITE_CHECK(TreeliteTreeBuilderCreateValue(
+          &q_node.node.quesval, TreeliteType<T>::value, &threshold));
         TREELITE_CHECK(TreeliteTreeBuilderSetNumericalTestNode(
           tree_builder, q_node.unique_node_id, q_node.node.colid,
-          "<=", q_node.node.quesval, 1, node_id + 1, node_id + 2));
+          "<=", threshold, 1, node_id + 1, node_id + 2));
+        TREELITE_CHECK(TreeliteTreeBuilderDeleteValue(threshold));
 
         node_id += 2;
       } else {
-        if (num_output_group == 1) {
+        if (num_class == 1) {
+          ValueHandle leaf_value;
+          if (std::is_same<L, int>::value) {
```

Comment:
Is int the only potentially problematic type, or would any integral type be an issue? If the latter, then [is_integral](https://en.cppreference.com/w/cpp/types/is_integral) will take care of this cleanly.

Reply:
Yes, the template `build_treelite_tree` has only four instantiations:
```cpp
template void build_treelite_tree<float, int>(
  TreeBuilderHandle tree_builder,
  DecisionTree::TreeMetaDataNode<float, int> *tree_ptr, int num_class);
template void build_treelite_tree<double, int>(
  TreeBuilderHandle tree_builder,
  DecisionTree::TreeMetaDataNode<double, int> *tree_ptr, int num_class);
template void build_treelite_tree<float, float>(
  TreeBuilderHandle tree_builder,
  DecisionTree::TreeMetaDataNode<float, float> *tree_ptr, int num_class);
template void build_treelite_tree<double, double>(
  TreeBuilderHandle tree_builder,
  DecisionTree::TreeMetaDataNode<double, double> *tree_ptr, int num_class);
```